### PR TITLE
use prelude

### PR DESCRIPTION
**Pull request by zapashcanon**

### CODE REVIEW

File: src/symbolic/symbolic_memory.ml [hunk 1]
```diff
@@ -154,7 +154,8 @@ let extract v pos =
     value (Num (I8 i'))
   | Cvtop (_, Zero_extend 24, ({ node = Symbol _; _ } as sym))
   | Cvtop (_, Sign_extend 24, ({ node = Symbol _; _ } as sym))
-    when ty sym = Ty_bitv 8 ->
+    when match ty sym with Ty_bitv 8 -> true | _ -> false ->
+    (* TODO: implement an equal function in smtml for this *)
```
Comment (zapashcanon) [inline src/symbolic/symbolic_memory.ml:158]: cc @filipeom, do you think you could add such a function ?

Comment (filipeom) [reply src/symbolic/symbolic_memory.ml:158]: A equality function for `Ty.t` ? We have [this](https://github.com/formalsec/smtml/blob/a3c524421082acf4c76e5dcc83e9ea133bdd2264/lib/ty.ml#L307), would it suffice or do we need something better?

File: src/cmd/cmd_conc.ml [hunk 2]
```diff
@@ -420,50 +418,58 @@ let cmd profiling debug unsafe optimize workers no_stop_at_failure no_values
...
 
   let testcase model =
     if not no_values then
       let testcase =
-        List.sort compare (Smtml.Model.get_bindings model) |> List.map snd
+        let compare_pair fx fy (x1, y1) (x2, y2) =
+          let cx = fx x1 x2 in
+          if cx = 0 then fy y1 y2 else cx
+        in
+        (* TODO: add a function for this in smtml *)
```
[the remark this answers is not in the document]
Comment (filipeom) [reply src/cmd/cmd_conc.ml:439]: Essentially move `compare_pair` into `Smtml.Model.compare_pair`? Yes, of course :smiley:

[the remark this answers is not in the document]
Comment (filipeom) [reply src/cmd/cmd_conc.ml:439]: I just would like to remark, that I wouldn't see a case where we would have 2 equals symbols and would need to decide the order based on a `compare` with the value. Also, a while back you made `get_bindings` return an ordered list https://github.com/formalsec/smtml/blob/a3c524421082acf4c76e5dcc83e9ea133bdd2264/lib/model.ml#L29C1-L31C68

[the remark this answers is not in the document]
Comment (zapashcanon) [reply src/cmd/cmd_conc.ml:439]: Sorry, I was talking about the few lines after to compare bindings :D

File: src/symbolic/symbolic_memory.ml (hunk 1 above)
Comment (zapashcanon) [reply src/symbolic/symbolic_memory.ml:158]: Indeed.. Thanks!

**Pull request merged**